packets.go

```diff
@@ -96,6 +96,19 @@ func (mc *mysqlConn) writePacket(data []byte) error {
 		return ErrPktTooLarge
 	}
 
+	if mc.reset.IsSet() {
```

Comment:
Why do you check this?

I don't think this is needed.  Connections in pool may not be used for long time.  While waiting it is reused,
the connection may be closed from remote.

Reply:
HI! I've explained this in the commit message :smile: 

Inlined here for your convenience:

```
This check is not performed for all writes. Go 1.10 introduced a new
`sql/driver` interface called `driver.SessionResetter`, which calls the
`ResetSession` method on any connections _when they are returned to the
connection pool_. Since performing the liveness check during
`ResetSession` is not particularly useful (the connection can spend a
long time in the pool before it's checked out again, and become stale),
we simply mark the connection with a `reset` flag instead.

This `reset` flag is then checked from `mysqlConn.writePacket` to
perform the liveness checks. This ensures that the liveness check will
only be performed for the first query on a connection that has been
checked out of the connection pool. These are pretty much the semantics
we want: a fresh connection from the pool is more likely to be stale,
and it has not performed any previous writes that could cause data
corruption. If a connection is being consistently used by the client
(i.e. through an open transaction), we do NOT perform liveness checks.
If MySQL Server kills such active connection, we want to bubble up the
error to the user because any silent retrying can and will lead to data
corruption.
```

Reply:
Ah, I misunderstood.  Please leave some comment for it.

Anyway, no need to use atomic.  `database/sql` don't call APIs simultaneously, except `Rows.Close()`.

Reply:
Done!